eth_interface.md

```diff
@@ -6,7 +6,7 @@ The Ethereum Environment Interface exposes the core Ethereum API to the eWASM en
 
 We define the following Ethereum data types:
 - `bytes`: an array of bytes with unrestricted length
-- `address`: a 160 bit number, represented as a 20 bytes long little endian unsigned integer in memory
+- `address`: an array of 20 bytes
 - `u128`: a 128 bit number, represented as a 16 bytes long little endian unsigned integer in memory
 - `u256`: a 256 bit number, represented as a 32 bytes long little endian unsigned integer in memory
```

Comment:
No. I think that would not be sufficient. You would have to explain where these bytes come from, i.e. how numbers where serialized to bytes, i.e. endianness.

In case of address this is not needed because an address is a well specified part of a hash, where a hash is defined as bytes, not a number.

Reply:
u128 should be defined as an integer value, since it (usually?) denotes an ether value and native arithmetic and comparison operators should work on them in the WASM environment (by experimental u128 or otherwise).
u256 is very difficult because a hash value should not be byte-swapped but the difficulty should be, as it is a quantity.
Byte ordering is therefore highly dependent on context in the case of u256.

Reply:
So make two types.